backend/common/position.go

```diff
@@ -78,3 +78,18 @@ func ConvertTiDBParserErrorPositionToPosition(line, column int) *storepb.Positio
 		Column: int32(column),
 	}
 }
+
+// ConvertANTLRTokenToExclusiveEndPosition converts an ANTLR token position to an exclusive end Position.
+// The end token's line and column (0-based) plus its text are used to calculate where the position
+// should point to (after the last character of the token).
+// This is used for Statement.End which should use exclusive semantics (pointing after the last character).
+func ConvertANTLRTokenToExclusiveEndPosition(line int32, column int32, tokenText string) *storepb.Position {
+	tokenCharLength := int32(len([]rune(tokenText)))
```

Comment:
len(tokenText) instead of len([]rune(tokenText))?

bytes or chars?

Reply:
Character index, follow proto spec https://github.com/bytebase/bytebase/blob/main/proto/store/store/common.proto#L99